src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -138,6 +138,13 @@ protected void AddVariablesToEnvironment(bool excludeNames = false, bool exclude
                 string formattedKey = (pair.Key ?? string.Empty).Replace('.', '_').ToUpperInvariant();
                 AddEnvironmentVariable(formattedKey, pair.Value);
 
+                // for maintaining back compat variable names replace space with underscore
+                string backCompatKey = formattedKey.Replace(' ', '_');
```

Comment:
Agreed.  I talked over with some folks over here.  Just have one var and chain the replaces on line 138.  Agreed it's odd but our UI apparently lets you do it.